initialization/initialization.go

```diff
@@ -66,6 +60,7 @@ type option struct {
 
 	logger            *Logger
 	powDifficultyFunc func(uint64) []byte
+	referenceOracle   *oracle.WorkOracle
```

Comment:
NIT, how about using this: https://github.com/spacemeshos/post/blob/3df31e0357396010460ef7f1ff0a0ccedbff27ff/internal/postrs/api.go#L34-L37
```suggestion
	referenceOracle   Scrypter
```

Reply:
`WorkOracle` doesn't implement `Scrypter`, instead it uses an `Scrypter` (implemented by `postrs.Scrypt`) and extends it with some additional error checking. Otherwise I would use it 🙂